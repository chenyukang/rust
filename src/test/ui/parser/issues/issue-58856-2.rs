struct Empty;

trait Howness {}

impl Howness for () {
    fn how_are_you(&self -> Empty {
    //~^ ERROR expected parameter name
    //~^^ ERROR expected one of
    //~^^^ ERROR method `how_are_you` is not a member of trait `Howness`
    //~| ERROR associated function in `impl` without body
        Empty
    }
}
//~^ ERROR mismatched closing delimiter
//~| ERROR expected one of

fn main() {}
