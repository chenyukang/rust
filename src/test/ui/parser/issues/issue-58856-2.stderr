error: unexpected closing delimiter: `}`
  --> $DIR/issue-58856-2.rs:11:1
   |
LL |     fn how_are_you(&self -> Empty {
   |                                   - this opening brace...
...
LL |     }
   |     - ...matches this closing brace
LL | }
   | ^ unexpected closing delimiter

error: this file contains an unclosed delimiter
  --> $DIR/issue-58856-2.rs:14:14
   |
LL | impl Howness for () {
   |                     - unclosed delimiter
LL |     fn how_are_you(&self -> Empty {
   |                   - unclosed delimiter
...
LL | fn main() {}
   |              ^

error: this file contains an unclosed delimiter
  --> $DIR/issue-58856-2.rs:14:14
   |
LL | impl Howness for () {
   |                     - unclosed delimiter
LL |     fn how_are_you(&self -> Empty {
   |                   - unclosed delimiter
...
LL | fn main() {}
   |              ^

error: expected parameter name, found `->`
  --> $DIR/issue-58856-2.rs:6:26
   |
LL |     fn how_are_you(&self -> Empty {
   |                          ^^ expected parameter name

error: expected one of `)` or `,`, found `->`
  --> $DIR/issue-58856-2.rs:6:26
   |
LL |     fn how_are_you(&self -> Empty {
   |                         -^^ expected one of `)` or `,`
   |                         |
   |                         help: missing `,`

error: expected one of `->`, `where`, or `{`, found `}`
  --> $DIR/issue-58856-2.rs:14:14
   |
LL |     fn how_are_you(&self -> Empty {
   |        ----------- while parsing this `fn`
...
LL | fn main() {}
   |              ^ expected one of `->`, `where`, or `{`

error: associated function in `impl` without body
  --> $DIR/issue-58856-2.rs:6:5
   |
LL | /     fn how_are_you(&self -> Empty {
LL | |
LL | |
LL | |         Empty
...  |
LL | |
LL | | fn main() {}
   | |             ^- help: provide a definition for the function: `{ <body> }`
   | |_____________|
   | 

error[E0407]: method `how_are_you` is not a member of trait `Howness`
  --> $DIR/issue-58856-2.rs:6:5
   |
LL | /     fn how_are_you(&self -> Empty {
LL | |
LL | |
LL | |         Empty
...  |
LL | |
LL | | fn main() {}
   | |_____________^ not a member of trait `Howness`

error: aborting due to 8 previous errors

For more information about this error, try `rustc --explain E0407`.
